doc/mpls/MPLS_hld.md

```diff
@@ -0,0 +1,509 @@
+      - [Warm Boot Requirements](#warm-boot-requirements)
+      - [Future Requirements](#future-requirements)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Overview](#overview-1)
+      - [Database Changes](#database-changes)
+        - [APPL DB](#appl-db)
+          - [INTERFACE_TABLE](#interface_table)
+          - [ROUTE_TABLE](#route_table)
+          - [LABEL_ROUTE_TABLE](#label_route_table)
+      - [Software Modules](#software-modules)
+        - [NetLink](#netlink)
+          - [Functions](#functions)
+        - [IntfMgr](#intfmgr)
+        - [FPM Syncd](#fpm-syncd)
+          - [Functions](#functions-1)
+        - [IntfsOrch](#intfsorch)
+          - [Functions](#functions-2)
+        - [RouteOrch](#routeorch)
+          - [Functions](#functions-3)
+        - [NeighOrch](#neighorch)
+          - [Functions](#functions-4)
+        - [Label/LabelStack](#labellabelstack)
+        - [NextHopKey](#nexthopkey)
+        - [Syncd](#syncd)
+    - [SAI API](#sai-api)
+      - [Router Interface](#router-interface)
+      - [MPLS](#mpls)
+      - [Next Hop](#next-hop)
+      - [Next Hop Group](#next-hop-group)
+    - [Configuration and management](#configuration-and-management)
+      - [CLI Enhancements](#cli-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+        - [INTERFACE](#interface)
+        - [PORTCHANNEL_INTERFACE](#portchannel_interface)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+### Revision
+|  Rev  | Date  |  Author  | Change Description |
+| :---: | :---: | :------: | ------------------ |
+|  0.1  |       | A Pokora | Initial version    |
+
+### Scope
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Definitions/Abbreviations
+| Abbreviation | Description                           |
+| ------------ | ------------------------------------- |
+| cRPD         | Containerized Routing Protocol Daemon |
+| ECMP         | Equal Cost Multi-Path                 |
+| MPLS         | Multi-Label Packet Switching          |
+
+### Overview
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Requirements
+
+This section describes the SONiC requirements for:
+- MPLS routes feature.
+- Weighted ECMP routes feature.
+
+#### Functional requirements
+- Support to enable/disable MPLS per Interface.
+- Support for MPLS Push, Pop, and Swap routes.
+- Support for weighted ECMP routes for IPv4/IPv6 and MPLS.
+- Support for bulk MPLS route programming.
+- Integration with Juniper cRPD routing stack.
```

Comment:
Could move the Juniper integration to something like "testing done" instead of "requirements"